developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,247 @@
+   };
+
+| **auto_start**: Indicates whether a module instance of the module should be created at the base fw startup
+| **sched_caps**: It is a deprecated data used by driver to calculate schedule info for module. Please use the
+| default frame_length 1 and all multiple supported.
+
+.. code-block:: bash
+
+    struct SchedulingCapabilities {
+        uint16_t   frame_length; // Input frame length in sample groups
+        Multiples  multiples_supported;
+    };
+
+    struct Multiples {
+        uint16_t x1 : 1;   // (base * 1)
+        uint16_t x2 : 1;   // (base * 2)
+        uint16_t x3 : 1;   // (base * 3)
+        uint16_t x4 : 1;   // (base * 4)
+        uint16_t x5 : 1;   // (base * 5)
+        uint16_t x6 : 1;   // (base * 6)
+        uint16_t x7 : 1;   // (base * 7)
+        uint16_t x8 : 1;   // (base * 8)
+        uint16_t x9 : 1;   // (base * 9)
+        uint16_t x10 : 1;  // (base * 10)
+        uint16_t x11 : 1;  // (base * 11)
+        uint16_t x12 : 1;  // (base * 12)
+        uint16_t x13 : 1;  // (base * 13)
+        uint16_t x14 : 1;  // (base * 14)
+        uint16_t x15 : 1;  // (base * 15)
+        uint16_t all : 1;  // (base * X)
+    };
+
+| **pin**: is array of data used to define the capability of input & output
+
+.. code-block:: bash
+
+  pin = [dir, type, sample rates, sample size, sample container size, channel config]
+  For example, gain module pin = [0, 0, 0xfeef, 0xf, 0xf, 0x45ff], it supports input pcm
+  supported sample rate [0xxfeef]: 8000, 11500, 16000, 1200, 22050, 24000, 32000, 44100,
+                                   48000, 64000, 88200, 96000, 176400, 19200,
+  supported sample size [0xf]: 8bits, 16bits, 24bits, 32bits.
+  supported sample container size [0xa]: 16bits and 32bits
+  supported channel config [0x45ff]: mono, dual mono, stereo, 2_1, 3_0, quad, surround, 3_1,
+                                     5_0_surround, 7_1
+
+  enum dir {
+   input = 0,
+   output,
+  }
+
+  enum type {
+    pcm = 0,
+    mp3,
+    aac,
+  };
+
+  struct sample_rates {
+    uint32_t freq_8000   : 1;
+    uint32_t freq_11500  : 1;
+    uint32_t freq_12000  : 1;
+    uint32_t freq_16000  : 1;
+    uint32_t freq_18900  : 1;
+    uint32_t freq_22050  : 1;
+    uint32_t freq_24000  : 1;
+    uint32_t freq_32000  : 1;
+    uint32_t freq_37800  : 1;
+    uint32_t freq_44100  : 1;
+    uint32_t freq_48000  : 1;
+    uint32_t freq_64000  : 1;
+    uint32_t freq_88200  : 1;
+    uint32_t freq_96000  : 1;
+    uint32_t freq_176400 : 1;
+    uint32_t freq_192000 : 1;
+    uint32_t reserved_   : 16;
+  };
+
+  struct sample_sizes {
+    uint16_t bits_8    : 1;
+    uint16_t bits_16   : 1;
+    uint16_t bits_24   : 1;
+    uint16_t bits_32   : 1;
+    uint16_t bits_64   : 1;
+    uint16_t reserved_ : 11;
+  };
+
+  struct sample_containers {
+    uint16_t bits_8    : 1;
+    uint16_t bits_16   : 1;
+    uint16_t bits_32   : 1;
+    uint16_t bits_64   : 1;
+    uint16_t reserved_ : 12;
+  };
+
+  struct channel_configurations {
```

Comment:
Why does the FW (and/or the driver) need the channel configuration?

Reply:
@mmaka1  This is included in module extended config and we can use fw parser to decode fw binary to get it.  It is used by windows driver but I am not familiar with windows driver. Please check the following dump for copier module in windows fw.  We add it to make sure cSOF is compatible with window driver.

````
module b COPIER
uuid 9BA00C83-CA12-4A83-943C-1FA2E82F9DDA
name Copier
affinity_mask MASTER_CORE_AFFINITY
instance_count COPIER_MAX_INSTANCE
domain_types LL
type CopierModule
group text
section ep .copier.cmi.text


sched_caps 1 all


pin in
stream_type pcm
sample_rates 8k 11.025k 12k 16k 22.05k 24k 32k 44.1k 48k 64k 88.2k 96k 176.4k 192k
sample_sizes sample_8b sample_16b sample_24b sample_32b
sample_containers container_8b container_16b container_24b container_32b
channel_cfg ch_mono ch_dual_mono ch_stereo ch_2_1 ch_3_0 ch_3_1 ch_quad ch_surround ch_5_0 ch_5_1 ch_7_1

pin out
stream_type pcm
sample_rates 8k 11.025k 12k 16k 22.05k 24k 32k 44.1k 48k 64k 88.2k 96k 176.4k 192k
sample_sizes sample_8b sample_16b sample_24b sample_32b
sample_containers container_8b container_16b container_24b container_32b
channel_cfg ch_mono ch_dual_mono ch_stereo ch_2_1 ch_3_0 ch_3_1 ch_quad ch_surround ch_5_0 ch_5_1 ch_7_1

pin out
stream_type pcm
sample_rates 8k 11.025k 12k 16k 22.05k 24k 32k 44.1k 48k 64k 88.2k 96k 176.4k 192k
sample_sizes sample_8b sample_16b sample_24b sample_32b
sample_containers container_8b container_16b container_24b container_32b
channel_cfg ch_mono ch_dual_mono ch_stereo ch_2_1 ch_3_0 ch_3_1 ch_quad ch_surround ch_5_0 ch_5_1 ch_7_1

pin out
stream_type pcm
sample_rates 8k 11.025k 12k 16k 22.05k 24k 32k 44.1k 48k 64k 88.2k 96k 176.4k 192k
sample_sizes sample_8b sample_16b sample_24b sample_32b
sample_containers container_8b container_16b container_24b container_32b
channel_cfg ch_mono ch_dual_mono ch_stereo ch_2_1 ch_3_0 ch_3_1 ch_quad ch_surround ch_5_0 ch_5_1 ch_7_1

pin out
stream_type pcm
sample_rates 8k 11.025k 12k 16k 22.05k 24k 32k 44.1k 48k 64k 88.2k 96k 176.4k 192k
sample_sizes sample_8b sample_16b sample_24b sample_32b
sample_containers container_8b container_16b container_24b container_32b
channel_cfg ch_mono ch_dual_mono ch_stereo ch_2_1 ch_3_0 ch_3_1 ch_quad ch_surround ch_5_0 ch_5_1 ch_7_1
````